proto/cacheclient.proto

```diff
@@ -402,16 +402,89 @@ message _ListRemoveRequest {
 
 message _ListRemoveResponse {}
 
+// List fetch + retain index behavior
+// This follows redis behavior for the most part, differences
+// 1. end_index is exclusive, unless None
+// 2. `start_index` == 0, then we start at the front of list
+// 2. `end_index` == None, start at the end of the index, inclusive
+/*
+  list_ = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+  # Using the slice notation:
+  # start:stop(:step)?
+
+
+  # start=None, end=None
```

Comment:
nit: None == 0, I'd recommend just using `0` everywhere in here instead of also using the alias "none"

Reply:
Besides end_index, yeah? Start index can all just be 0, but for end index there's a difference between 0 and None, 0 being start of array and None being end of list.

for example, we have
```
  # start=None, end=None
  list_[:] == [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
```
and then 5:0
```
# start=5, end=0
  list_[5:0] == []
```

but what about, 0:0
```
# start=0, end=0
  list_[0:0] == []
```
or do we change end index being exclusive to inclusive for 0 values?
```
# start=0, end=0
  list_[0:0] == [0]
```

Reply:
Actually, this one case, 0,0 is a real problem. Since we treat the end index as exclusive, our logic says
```
# start=0, end=0
  list_[0:0] == []
```

but redis says
```
# start=0, end=0
  list_[0:0] == [0]
```

so how are we suppose to model this in ms2? Only options here is to,
1. In our docs we say that end_index is exclusive, unless 0 or omitted, if 0, then it includes the beginning of the list, if omitted, goes to end of list, this option actually makes sense and keeps it consistent with how we handle start/end list boundaries
2. in ms2, if we meet the case where start and end index is 0, increment start_index so that start is greater then end, causing redis to return a empty array [] 😩 👎 
```
# start=0, end=0
  list_[1:0] == []
```

Reply:
@malandis thoughts?